programs/cryptid/src/instructions/execute_transaction.rs

```diff
@@ -12,19 +12,19 @@ use anchor_lang::prelude::*;
 /// A vector of the number of extras for each signer, signer count is the length
 controller_chain: Vec<u8>,
 /// The bump seed for the Cryptid signer
-bump: u8,
+cryptid_account_bump: u8,
+/// Index of the cryptid account
+cryptid_account_index: u32,
+/// The bump seed for the Did Account
+did_account_bump: u8,
 /// Additional flags
 flags: u8,
 )]
 pub struct ExecuteTransaction<'info> {
     /// The Cryptid instance to execute with
-    /// CHECK: This assumes a purely generative case until we have use-cases that require a state.
-    #[account(
-        mut,
-        seeds = [CryptidAccount::SEED_PREFIX, did_program.key().as_ref(), did.key().as_ref(), cryptid_account.index.to_le_bytes().as_ref()],
-        bump
-    )]
-    pub cryptid_account: Account<'info, CryptidAccount>, // TODO use new macro that allows generative accounts
+    /// CHECK: Cryptid Account can be generative and non-generative
```

Comment:
I suggest putting the macro checks back in but commented out with a comment saying to use them when the generative macro is available.